homeassistant/components/shelly/sensor.py

```diff
@@ -373,6 +387,30 @@ class RestSensorDescription(RestEntityDescription, SensorEntityDescription):
         entity_category=EntityCategory.DIAGNOSTIC,
         use_polling_wrapper=True,
     ),
+    "humidity_0": RpcSensorDescription(
+        key="humidity:0",
+        sub_key="rh",
+        name="Humidity",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.HUMIDITY,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_registry_enabled_default=True,
+    ),
+    "battery": RpcSensorDescription(
+        key="devicepower:0",
+        sub_key="battery",
+        name="Battery",
+        native_unit_of_measurement=PERCENTAGE,
+        value=lambda status, _: round(status["percent"], 2),
+        extra_state_attributes=lambda status, shelly: {
+            "Voltage": cast(int, status["V"])
+        },
```

Comment:
Hi @chemelli74.
Home Assistant recommendation is to avoid using `extra_state_attributes` for sensors, I would say even not allow it. We can add another sensor "Battery Voltage" if you want.
```suggestion
```

Reply:
We used `extra_state_attributes` for other sensors, this is why I though it was not worth to have a separate sensor.

Reply:
Ok, let's go with it, we can migrate it to a sensor in the future if we are requested as it is a diagnostic sensor anyhow, but it needs to be float.
```suggestion
        extra_state_attributes=lambda status, shelly: {
            "Voltage": round(status["V"], 2)
        },
```